mlflow/types/llm.py

```diff
@@ -0,0 +1,306 @@
+        self._convert_dataclass_list("messages", ChatMessage)
+        super().__post_init__()
+
+
+@dataclass
+class ChatChoice(_BaseDataclass):
+    """
+    A single chat response generated by the model.
+
+    :param index:         The index of the response in the list of responses.
+    :type index:          int
+    :param message:       The message that was generated.
+    :type message:        :py:class:`ChatMessage`
+    :param finish_reason: The reason why generation stopped.
+    :type finish_reason:  str
+    """
+
+    index: int
+    message: ChatMessage
+    finish_reason: str
+
+    def __post_init__(self):
+        self._validate_field("index", int, True)
+        self._validate_field("finish_reason", str, True)
+        if not isinstance(self.message, ChatMessage):
+            self.message = ChatMessage(**self.message)
+
+
+@dataclass
+class TokenUsageStats(_BaseDataclass):
+    """
+    Stats about the number of tokens used during inference.
+
+    :param prompt_tokens:     The number of tokens in the prompt.
+    :type prompt_tokens:      int
+    :param completion_tokens: The number of tokens in the generated completion.
+    :type completion_tokens:  int
+    :param total_tokens:      The total number of tokens used.
+    :type total_tokens:       int
+    """
+
+    prompt_tokens: int
+    completion_tokens: int
+    total_tokens: int
+
+    def __post_init__(self):
+        self._validate_field("prompt_tokens", int, True)
+        self._validate_field("completion_tokens", int, True)
+        self._validate_field("total_tokens", int, True)
+
+
+@dataclass
+class ChatResponse(_BaseDataclass):
+    """
+    The full response object returned by the chat endpoint.
+
+    :param id:      The ID of the response.
+    :type id:       str
+    :param object:  The object type.
+    :type object:   str
+    :param created: The time the response was created. **Optional**, defaults to the current time.
+    :type created:  int
+    :param model:   The name of the model used.
+    :type model:    str
+    :param choices: A list of :py:class:`ChatChoice` objects containing the
+                    generated responses
+    :type choices:  List[:py:class:`ChatChoice`]
+    :param usage:   An object describing the tokens used by the request.
+    :type usage:    :py:class:`TokenUsageStats`
+    """
+
+    id: str
+    model: str
+    choices: List[ChatChoice]
+    usage: TokenUsageStats
+    object: Literal["chat.completion"] = "chat.completion"
+    created: int = field(default_factory=lambda: int(time.time()))
+
+    def __post_init__(self):
+        self._validate_field("id", str, True)
+        self._validate_field("object", str, True)
+        self._validate_field("created", int, True)
+        self._validate_field("model", str, True)
+        self._convert_dataclass_list("choices", ChatChoice)
+        if not isinstance(self.usage, TokenUsageStats):
+            self.usage = TokenUsageStats(**self.usage)
```

Comment:
same here

Reply:
changed this to check for `dict` and throw `ValueError` after if the field is not an instance of the expected type